p2p/sender.go

```diff
@@ -26,18 +27,78 @@ import (
 
 	"github.com/obolnetwork/charon/app/errors"
 	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
 )
 
-// SendAsync sends a libp2p message and logs a warning on error.
-//   Usage: go p2p.SendAsync(ctx, tcpNode, protoId, peerId, msg)
-func SendAsync(ctx context.Context, tcpNode host.Host, protoID protocol.ID, peerID peer.ID, msg proto.Message) {
-	err := sendMsg(ctx, tcpNode, protoID, peerID, msg)
-	if err != nil {
-		log.Warn(ctx, "Failed sending p2p message", err)
+const (
+	senderHysteresis = 3
+	senderBuffer     = senderHysteresis + 1
+)
+
+type peerState struct {
+	failing bool
+	buffer  []error
```

Comment:
nit: can rename to `bufErrs`

Reply:
I prefer to call the thing by what it is/does, rather than adding type suffixes, especially if it is there is only one of them